src/encode.py

```diff
@@ -0,0 +1,45 @@
+import pandas as pd
+
+FEATURES_DEFAULT = ('lesson_id', 'exercise_code', 'learning_object', 'exercise_code_level')
+TIME_WINDOWS_DEFAULT = ('60s', '1h', '1d', '5d', '30d', '365d')
+COUNTERS_DEFAULT = ('attempts', 'wins')
+
+def encode_df(df, feature_cumcount=[]):
+    '''
+    Encodes the wins and fails cumulative counts in the given time windows and dummifies the categorical variables of the dataset.
```

Comment:
I think the function should not do 2 things (like cumulative counts and dummify) if we want to respect the SRP (but it's chipotage)

Reply:
corrected